impeller/entity/entity_unittests.cc

```diff
@@ -1751,6 +1753,7 @@ TEST_P(EntityTest, RuntimeEffect) {
 
     ASSERT_TRUE(runtime_stage->IsDirty());
     expect_dirty = true;
+    first_pipeline = PipelineRef{nullptr};
```

Comment:
Changing this test to null out the pipeline. The test is clearing out the shader cache which invalidates this ptr.

Reply:
I'm not sure if this is the right fix.  The reason being that the failure was `Expected: (first_pipeline) != (pass.GetCommands().back().pipeline)`.  It's an assertion that doesn't even look into the validity of the pointer, it just performs a computation on the address.  Were you able to validate locally this removed the failure?

Reply:
I'm wary of changing this test to fix the failure.  I'm afraid we may be masking a bug.  My intuition on this assertion is that the render pass's commands should point to a different pipeline if the runtime stage has been updated.  That makes sense.  This change just removes that assertion by just asserting that the pipeline is not null.  So this probably removes the flakiness but not the bug it's demonstrating.

Reply:
I don't really think that this is a meaningful assertion

Reply:
Chatted offline: The most likely reason that sometimes this fails is that we get (un)lucky and we're creating a new pipeline that has the same address.  Jonah's right that this is not a good assertion.  I'd say that usually `not equals` assertions are kind of trash anyways.  We should have some sort of positive assertion about the rebuild having happened.

Reply:
I can't come up with a better alternative to this assert.  This is fine for me.  Although I think it's better to just delete that `EXPECT_NE` or replace it with `EXPECT_NE(nullptr, pass.GetCommands().back().pipeline.get());` instead of fiddling with state.